### ISSUE TITLE
Android RuntimeException with LocationProviderChangedReceiver

### ISSUE THREAD

**Issue by sebastien-roch** [dpa99c/cordova-diagnostic-plugin#80]
Hi,
thanks for the nice plugin. Unfortunately it crashes on my device with the following error and stacktrace:
```
java.lang.RuntimeException: Unable to start receiver cordova.plugins.Diagnostic$LocationProviderChangedReceiver: java.lang.NullPointerException: Attempt to invoke virtual method 'void cordova.plugins.Diagnostic.notifyLocationStateChange()' on a null object reference
E/AndroidRuntime(21471): 	at android.app.ActivityThread.handleReceiver(ActivityThread.java:2591)
E/AndroidRuntime(21471): 	at android.app.ActivityThread.access$1700(ActivityThread.java:148)
E/AndroidRuntime(21471): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1360)
etc...
```
It seems that on line 837, `instance` is null. Don't know if it's always null or only sometimes.
I think the thing is fixable by registering the BroadcastReceiver in-line (and not through the manifest) so that we can keep a reference of the CordovaPlugin. But the question is more: why am I the only one encountering this bug?

Any idea? thanks
Android version 5.0, Cordova 6.2, plugin freshly installed.

**Comment by parham-fazel** [comment]
You are not the only one having this issue, I have the exact same problem!
I think your solution can fix it.

**Comment by tinderangler** [comment]
Tried that today, same problem for me.

**Comment by tinderangler** [comment]
Did some testing: The problem is that `initialize()` only gets called (and thus `instance` set) when interacting with the Java component of the plugin. `registerLocationStateChangeHandler()` is only interacting with the JavaScript part and then `instance` remains unset.

As a workaround I'm now just calling one other function to initialize the plugin and everything is so far looking well.

**Comment by nicoabie** [comment]
I haven't tested this but a fix would be loading the plugin as soon as the application launches.
Cordova has a single instance of every plugin but gets initialized the first time the user calls it from javascript.

So to have it initialized on load time we should create a pull request (not having time today) changing 

```xml
<feature name="Diagnostic" >
<param name="android-package" value="cordova.plugins.Diagnostic"/>
</feature>
```

to

```xml
<feature name="Diagnostic" >
<param name="android-package" value="cordova.plugins.Diagnostic"/>
<param name="onload" value="true" />
</feature>
```

in the plugin manifest.

**Comment by dpa99c** [comment]
Sorry it's taken a while - just got back from holiday.

This is resolved by commits 44107620961b1a93c6e63ea3463ca768a81ea320 and 92f64dec91b6e11fb495ab5ed105e173b3ea5ba3 and published to npm as `cordova.plugins.diagnostic@3.0.3`

**Comment by dpa99c** [comment]
Note that `cordova.plugins.diagnostic@3.0.4` supersedes this to fix #85.
